components/HomeScreen/ContinentsList/ContinentsList.jsx

```diff
@@ -4,30 +4,30 @@ import { useRouter } from "expo-router";
 
 
 /**
- * ContinentsList.
+ * List of continents as pressables
  *
  * @param {object} props
- * @param {object} props.data
- * @param {boolean} props.loading
- * @param {function} props.refetch
+ * @param {object} [props.data]
+ * @param {"LOADING"|"ERROR"|"OK"} [props.responseStatus = "OK"]
  */
 function ContinentsList(props) {
 
   const {
-    data = {},
-    error = false,
-    loading = false,
-    refetch,
+    data,
+    responseStatus = "OK",
   } = props;
 
-  const isLoading = loading;
-
-  const isError = !isLoading && error;
-
-  const isOk = !isLoading && !isError;
-
   const router = useRouter();
 
+  /**
+   * creates a function to be called in the continent Pressable
+   *
+   * @param {string} code
+   */
+  function makeDoNavigate(code) {
```

Comment:
recommend pulling this into a component specific util file, both to clean up the component and make it clearer when something is business logic vs displaying